pkg/rancher-desktop/integrations/windowsIntegrationManager.ts

```diff
@@ -122,6 +141,30 @@ export default class WindowsIntegrationManager implements IntegrationManager {
   }
 
   async sync(): Promise<void> {
+    const latch = Latch();
+
+    switch (this.syncState.state) {
+    case SyncStateKey.IDLE:
+      this.syncState = { state: SyncStateKey.ACTIVE, active: latch };
+      break;
+    case SyncStateKey.ACTIVE: {
+      // There is a sync already active; wait for it, then do the re-sync.
+      const { active } = this.syncState;
+
+      this.syncState = {
+        state: SyncStateKey.QUEUED, active, queued: latch,
+      };
+      console.debug('Waiting for previous sync to finish before starting new sync.');
+      await active;
+      // Continue with the rest of the function, in ACTIVE mode.
+      break;
```

Comment:
It doesn't really matter, but why does this not simply `return this.syncState.queue`? Would it have the same effect?

Reply:
Because in that case nobody would actually schedule the run, so `.queued` would never resolve.